internal/cmd/perfcomp/energystatistics.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) MongoDB, Inc. 2025-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// Given two matrices, this function returns
+// (e, t, h) = (E-statistic, test statistic, e-coefficient of inhomogeneity)
+func GetEnergyStatistics(x, y *mat.Dense) (float64, float64, float64) {
```

Comment:
suggest unexporting all functions and moving them to main.go so that we can run using `go run main.go` instead of `go run .`.